Remove jekyll documentation

msecret opened the pull request:
Also makes an effort to update and improve the project documentation
by separating contributing information from install/use information
into a separate contributing file.

Refs https://github.com/18F/cg-style/issues/178

Commit:
Remove jekyll documentation

Also makes an effort to update and improve the project documentation
by separating contributing information from install/use information
into a separate contributing file.

adborden commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,60 @@
+0. If using **nvm**, run `nvm use`. Otherwise, try and install the node version that is in `.nvmrc`.
+0. Run `npm install` in the `cloudgov-style` repo.
+0. Run `npm run build` in the `cloudgov-style` repo.
+0. In the *cloudgov-style* repo, link the current module by running `npm link` in the root of the repo.
```
I might re-phrase this as
> In the *cloudgov-style* repo, make the module linkable by running `npm link` in the root of the repo.

I think it's good to have a distinction in terminology when talking about the app being linked vs app linking from.

adborden commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,60 @@
+   - Confirm that the cg-style folder exists in `node_modules` folder.
+
+0. In the *cloudgov-style* repo, watch the files and continually build by running `npm install && npm run watch`.
+0. If developing cg-style with either `cg-dashboard` or `cg-site`, run `npm run watch` in that project directory.
```
I like the consistency of referring to cg-style as cloudgov-style. We should try to do that throughout the doc.

adborden commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,60 @@
+<a name="development" id="development"></a>
+## Development and contributing setup
+
+These instructions explain how to develop the cloudgov style while using it on another (companion) site. This allows to see the result of the style changes on an actual site.
```
The distinction between cg-style and cloudgov-style isn't 100% clear, I think we could clarify that here. Maybe say:
> These instructions explain how to develop the cg-style (cloudgov-style) while using it on another (companion) site. This allows to see the result of the style changes on an actual site.

adborden commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,60 @@
+
+Run `npm run lint` to lint the codebase for style problems.
+
+When developing new components for the style library, ensure you add component documentation in `docs/src/components`.
```
`s/style library/pattern library/` please.

msecret replied:
I actually need to discuss that, I think it should actually be design system, not pattern library.

adborden commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,60 @@
+When developing new components for the style library, ensure you add component documentation in `docs/src/components`.
+
+## [WIP] Running cloud.gov pattern library locally
+The Cloud.gov pattern library leverages Fractal, a static site framework for building and documenting component libraries. Here are some instructions to view the library locally:
```
How about "building and documenting UI patterns." ?

adborden commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,60 @@
+
+```sh
+npm run build-library
+```
````
Technically, `npm run start` already does this step... But I think it's useful to keep it here in the docs so folks are aware the command exists.

adborden commented on package.json:
```diff
@@ -23,6 +23,7 @@
     "svg": "gulp --gulpfile build/gulpfile.js svg-sprite",
     "test": "npm run lint",
     "watch": "watch 'npm run build' ./src -d",
+    "watch-library": "watch 'npm run build-library-assets' ./src -d",
```
Thanks! This fixes https://github.com/18F/cg-style/issues/202

I also have an approach [here](https://github.com/18F/cg-style/compare/ab/edit-app-limit#diff-b9cfc7f2cdf78a7f4b91a753d10865a2) which ties into the `npm run build` command. If you're developing a new UI component in the library, you're probably already running `npm run watch`, so I think we should tie into that, otherwise you have to run two watches.

msecret replied:
I like that solution too

adborden commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,60 @@
+npm run watch-library
+```
+
+Go to `localhost:3000` in your browser and you should be viewing a local instance of the Cloud.gov pattern library. The `--sync` flag indicates that you would like Fractal to watch any changes you make. When your file is saved, Fractal will reload the site automatically and you will be able to view those changes.
````
Let's ditch the comment about `--sync`, it's confusing unless you're familiar with how npm commands work.

adborden commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,60 @@
+
+## Coding style
+
+There is a coding standards document that dictates standards and suggestions for how to write code: [documentation/frontend_standards.md](documentation/frontend_standards.md).
```
I think we just deleted this file, maybe move this content into CONTRIBUTING?

adborden reviewed (changes requested):
Sweet! Have a few comments/suggestions.

adborden commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,60 @@
+
+When developing new components for the style library, ensure you add component documentation in `docs/src/components`.
+
+## [WIP] Running cloud.gov pattern library locally
```
Let's remove the "[WIP]" part, since this is now the living documentation.

adborden commented:
I know some of my comments are referring to content you're just moving around, rather than creating, but it would be nice to clean up the docs on this pass too.

Commit:
Move standards into contributing and fixes

We removed the documentation where the front end coding standards
were, so moving into contributing where they belong. Also made
content/consistency updates.

Commit:
Modify and add some docs on new rules

adborden commented:
Quoting Marco Segreto (2016-12-09 10:45:04)
> msecret commented on this pull request.
>
> I actually need to discuss that, I think it should actually be design system, not pattern library.

Design system sounds fine to me, @dano?

--
Aaron D Borden
Human and Hacker

adborden commented on CONTRIBUTING.md:
```diff
@@ -2,40 +2,34 @@
 
-## [WIP] Running cloud.gov pattern library locally
-The Cloud.gov pattern library leverages Fractal, a static site framework for building and documenting component libraries. Here are some instructions to view the library locally:
+The Cloud.gov pattern library leverages Fractal, a static site framework for building and documenting component libraries. Here are some instructions for building and documenting UI patterns :
```
Oops, sorry I didn't catch this before. Let's link to http://fractal.build/

adborden commented on CONTRIBUTING.md:
```diff
@@ -55,6 +49,91 @@ To see changes as you're developing, ensure you're watching for file changes and
 
 Happy developing!
+
+## Frontend coding standards
```
:+1: this is all helpful stuff. I would probably like a little more detail on some of these rules, like _why_ is `@extend` not preferred in SASS? I'm also not sure if
